bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java

```diff
@@ -968,6 +1034,28 @@ public void readLastConfirmedDataComplete(int rc, DigestManager.RecoveryData dat
         new TryReadLastConfirmedOp(this, innercb, getLastAddConfirmed()).initiate();
     }
 
+    @Override
+    public CompletableFuture<Long> tryReadLastAddConfirmed() {
+        CompletableFuture<Long> result = new CompletableFuture<>();
+
+        ReadLastConfirmedCallback callback = (int rc, long lastConfirmed, Object ctx) -> {
+            SyncCallbackUtils.finish(rc, lastConfirmed, result);
+        };
+        asyncTryReadLastConfirmed(callback, result);
+        return result;
+    }
+
+    @Override
+    public CompletableFuture<Long> readLastAddConfirmed() {
+        CompletableFuture<Long> result = new CompletableFuture<>();
+
+        ReadLastConfirmedCallback callback = (int rc, long lastConfirmed, Object ctx) -> {
```

Comment:
ReadLastConfirmedCallback in line 1041 and line 1053 seem to be same. Can we follow the same pattern as how we handle other callbacks?